rtloader/two/two.cpp

```diff
@@ -46,20 +46,21 @@ Two::~Two()
     // For more information on why Py_Finalize() isn't called here please
     // refer to the header file or the doxygen documentation.
     PyEval_RestoreThread(_threadState);
-    free(_pythonHome);
     Py_XDECREF(_baseClass);
 }
 
 void Two::initPythonHome(const char *pythonHome)
 {
-    free(_pythonHome);
+    char *oldPythonHome = _pythonHome;
```

Comment:
I don't get the purpose of this change with the tmp var? Next lines, in all cases, we're re-assigning `_pythonHome`, so why bother keeping a pointer to free it at the end of the method? I'm surely missing something has you've did the same on purpose in `three.cpp`

Reply:
Ha ok got it, it's in case of the python runtime already using it? In order to support multiple initializations?

Reply:
Ha ok got it, it's in case of the python runtime already using it? In order to support multiple initializations?

Reply:
Yes, to free it *after* setting the new one (in the very unlikely case some other python thread accesses it after it has been freed, and the memory is not there anymore).